pages/[envId]/articles/[slug].tsx

```diff
@@ -66,16 +67,24 @@ const ArticlePage: FC<Props> = props => {
   );
 };
 
-export const getStaticProps: GetStaticProps<Props, { slug: string }> = async context => {
-  const siteMenu = await getSiteMenu(!!context.preview);
+export const getStaticProps: GetStaticProps<Props, { slug: string, envId: string }> = async context => {
+  const envId = context.params?.envId;
+  if (!envId) {
+    throw new Error("Missing envId in url");
+  }
+  const previewApiKey = context.previewData && typeof context.previewData === 'object' && 'currentPreviewApiKey' in context.previewData
+    ? context.previewData.currentPreviewApiKey as string
+    : undefined;
+
+  const siteMenu = await getSiteMenu({envId: envId, previewApiKey: previewApiKey}, !!context.preview);
   const slug = typeof context.params?.slug === "string" ? context.params.slug : "";
 
   if (!slug) {
     return { notFound: true };
   }
 
-  const article = await getArticleBySlug(slug, !!context.preview);
-  const defaultMetadata = await getDefaultMetadata(!!context.preview);
+  const article = await getArticleBySlug({envId: envId, previewApiKey: previewApiKey},slug, !!context.preview);
+  const defaultMetadata = await getDefaultMetadata({envId: envId, previewApiKey: previewApiKey}, !!context.preview);

```

Comment:
nit: You can shorten this `{envId: envId, previewApiKey: previewApiKey}` into this `{envId, previewApiKey}`.